xcodeproject/xcworkspace/schemes.go

```diff
@@ -44,30 +44,30 @@ func (w Workspace) Schemes() (map[string][]xcscheme.Scheme, error) {
 	// project schemes
 	projectLocations, err := w.ProjectFileLocations()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to get project locations from workspace: %w", err)
 	}
 
 	isAutocreateSchemesEnabled, err := w.isAutocreateSchemesEnabled()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to read the workspace autocreate scheme option: %w", err)
 	}
 
 	for _, projectLocation := range projectLocations {
 		if exist, err := pathutil.IsPathExists(projectLocation); err != nil {
-			return nil, fmt.Errorf("failed to check if project exist at: %s, error: %s", projectLocation, err)
+			return nil, fmt.Errorf("failed to check if project exist at: %s, error: %w", projectLocation, err)
 		} else if !exist {
 			// at this point we are interested the schemes visible for the workspace
 			continue
 		}
 
 		project, err := xcodeproj.Open(projectLocation)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to open project (%s): %w", projectLocation, err)
 		}
 
 		projectSchemes, err := project.SchemesWithAutocreateEnabled(isAutocreateSchemesEnabled)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to read project schemes: %w", err)
```

Comment:
I suggest including the project path in the error message, so that we know which project has the issue.
I would also unify how the current project path presents in these errors, we have embedded project related errors in line 57, 65 and 70 and they all present the project path in a different way.